sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/MethodDeclarationTracker.cs

```diff
@@ -55,20 +56,40 @@ void TrackMethodDeclaration(SymbolAnalysisContext c)
                     foreach (var declaration in c.Symbol.DeclaringSyntaxReferences)
                     {
                         var methodIdentifier = GetMethodIdentifier(declaration.GetSyntax());
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, methodIdentifier.GetLocation()));
+                        if (methodIdentifier.HasValue)
+                        {
+                            c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, methodIdentifier.Value.GetLocation()));
+                        }
                     }
                 }
             }
 
             bool IsTrackedMethod(IMethodSymbol methodSymbol, Compilation compilation)
             {
+                if (methodSymbol.MethodKind != MethodKind.Ordinary &&
+                    methodSymbol.MethodKind != MethodKind.ReducedExtension) // Just methods
```

Comment:
Why not make these conditions? This seems like quite a big constraint to bake into the base tracker class.

Also, the _MethodKind_ enum has values for _Constructor_, _Desctructor_, _PropertyGet_, _PropertySet_ etc.  Presumably this check is filtering out all of those types so the changes above to handle those kinds of method symbols aren't required if this check is done?

Reply:
I added logic to obtain identifier from all method symbol kinds and extracted the method kind check as a condition.